runtime/doc/lua.txt

```diff
@@ -574,6 +574,26 @@ retained for the lifetime of a buffer but this is subject to change. A plugin
 should keep a reference to the parser object as long as it wants incremental
 updates.
 
+Parser files						*treesitter-parsers*
+
+Parsers are the heart of tree-sitter. They are shared objects (`.so`) that
+tree-sitter will search for in the `parsers` runtime directory.
+
+For a parser to be available for a given language, there must be a file named
+`{lang}.so` within the parser directory.
+
+Query files						*treesitter-query*
+
+Queries are objects that tree-sitter uses to extract informations. They are
+built from a `scheme`-like language, and treesitter parses them to build the
+object.
+
+Treesitter also searches for query files within the `queries` directory. For a
+given language, treesitter will search the queries in the `queries/{lang}`. A
+query file is a `scheme` file (ending with `.scm`).
```

Comment:
so the query file must have `.scm` extension?

Reply:
Yep

Reply:
`queries/{lang}.scm` would be more consistent with the rest of the doc.

Reply:
That is not how it works, it is `queires/{Lang}/{query_name}.scm`.
I will document that better.